detox/src/devices/DeviceRegistry.js

```diff
@@ -4,25 +4,35 @@ const fs = require('fs-extra');
 const ExclusiveLockfile = require('../utils/ExclusiveLockfile');
 const safeAsync = require('../utils/safeAsync');
 
-const getDeviceEqualsFn = (deviceHandle) => (otherDeviceHandle) => _.isEqual(otherDeviceHandle, deviceHandle);
-const getDeviceDifferFn = (deviceHandle) => (otherDeviceHandle) => !_.isEqual(otherDeviceHandle, deviceHandle);
 const readOptions = {
   encoding: 'utf8',
 };
 
-class DeviceHandlesList {
+const FIELD_NAME_ID = 'id';
+const FIELD_NAME_DATA = 'data';
+
+class DevicesList {
```

Comment:
Overall, why, if you are already using ID as a primary factor, why don't you just turn the list into map, i.e.:

```
[id, meta],
[id, meta],
[id, meta],
```

to 

```
id: meta,
id: meta,
id: meta,
```

?

Reply:
I was trying to avoid the place where in the common use case we'd have a file containing a JSON with many keys and empty objects as data:
```json
{
  "id1": {},
  "id2: {},
   ...
}
```

Do you think it'd be better?